estimagic/optimization/tao_optimizers.py

```diff
@@ -0,0 +1,307 @@
+    and therefore may require fewer iterations to arrive at a local optimum than
+    Nelder-Mead.
+
+    The criterion function :func:`func` should return an array of the errors NOT an
+    array of the squared errors or anything else.
+
+    Scaling the problem is necessary such that bounds correspond to the unit hypercube
+    :mat:`[0, 1]^n`. For unconstrained problems, scale each parameter such that unit
+    changes in parameters result in similar order-of-magnitude changes in the criterion
+    value(s).
+
+    POUNDERs has several convergence criteria. Let :math:`X` be the current parameter
+    vector, :math:`X_0` the initial parameter vector, :math:`g` the gradient, and
+    :math:`f` the criterion function.
+
+    ``gradient_absolute_tolerance`` stops the optimization if the norm of the gradient
+    falls below :math:`\epsilon`.
+
+    .. math::
+
+        ||g(X)|| < \epsilon
+
+    ``gradient_relative_tolerance`` stops the optimization if the norm of the gradient
+    relative to the criterion value falls below :math:`epsilon`.
+
+    .. math::
+
+        ||g(X)|| / |f(X)| < \epsilon
+
+    ``gradient_total_tolerance`` stops the optimization if the norm of the gradient is
+    lower than some fraction :math:`epsilon` of the norm of the gradient at the initial
+    parameters.
+
+    .. math::
+
+        ||g(X)|| / ||g(X0)|| < \epsilon
+
+    Args:
+        gradient_absolute_tolerance (float): Stop if relative norm of gradient is less
+            than this. If set to False the algorithm will not consider
+            gradient_absolute_tolerance. Default is 1e-8.
+        gradient_relative_tolerance (float): Stop if norm of gradient is less than this.
+            If set to False the algorithm will not consider gradient_relative_tolerance.
+            Default is 1e-8.
+        gradient_total_tolerance (float): Stop if norm of gradient is reduced by this
+            factor. If set to False the algorithm will not consider
+            gradient_relative_tolerance. Default is 1e-10.
+        initial_trust_region_radius (float): Sets the radius for the initial trust
+            region that the optimizer employs. It must be :math:`> 0`.
+        max_iterations (int): Alternative Stopping criterion. If set the routine will
+            stop after the number of specified iterations or after the step size is
+            sufficiently small. If the variable is set the default criteria will all be
+            ignored. Default is `None`.
+
+    Returns:
+        results (dict): Dictionary with processed optimization results.
+
+    References:
+    .. _TAO Users Manual (Revision 3.7):
+        http://web.mit.edu/tao-petsc_v3.7/tao_manual.pdf
+    .. _Solving Derivative-Free Nonlinear Least Squares Problems with POUNDERS:
+        https://www.mcs.anl.gov/papers/P5120-0414.pdf
+    .. _petsc4py on BitBucket:
+        https://bitbucket.org/petsc/petsc4py
+
+    """
+    if sys.platform == "win32":
```

Comment:
We should replace this by a check that petsc4py is installed and have installation instructions in the error message.

Reply:
Would you like to collect all `is_xxx_installed` checks in the configuration file? The variable would be necessary in the optimizer packages and the tests and thus we could have a central place for it.

Reply:
Yes!